checkout/views.py

```diff
@@ -0,0 +1,150 @@
+from django.shortcuts import render, redirect, reverse, get_object_or_404
+from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+from .forms import MakePaymentForm, OrderForm
+from .models import OrderInfo, OrderLineItem
+from django.conf import settings
+from django.utils import timezone
+from products.models import ProductVariant
+from commissions.models import Quote
+import stripe
+
+import os
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+
+logger = logging.getLogger(__name__)
+
+# Create your views here.
+
+stripe.api_key = settings.STRIPE_SECRET_KEY
+
+@login_required()
+def checkout(request):
+    if request.method == "POST":
+        order_form = OrderForm(request.POST)
+        payment_form = MakePaymentForm(request.POST)
+        
+        if order_form.is_valid() and payment_form.is_valid():
+            order_info = order_form.save(commit=False)
+            order_info.customer_id = request.user.id
+            order_info.date = timezone.now()
+            order_info.save()
+            
+            cart = request.session.get('cart', {})
```

Comment:
add comment on what you're doing here

Reply:
Comments added. Empty object removed. Originally the checkout page still rendered even if the cart was empty but this behaviour is now prohibited.